Cargo.toml

```diff
@@ -26,7 +26,7 @@ z3 = "0.10.0"
 rustc-hash = "1"
 symbolic_expressions = "5"
 rayon = "1"
-smallvec = { version = "1.6", features = ["union", "const_generics"] }
+smallstr = "0.3"
```

Comment:
what's the advantage of smallstr over smallvec again?

Reply:
I think it leads to clearer code than using a smallvec of bytes since the data type is actually meant for strings. It also allows us to get rid of this [unsafe code block](https://github.com/uwplse/ruler/blob/main/src/bin/str.rs#L36)

@mwillsey can probably give a more detailed answer though!

Reply:
Yeah, smallstr makes sense to me.